Tests/Common/Securities/OptionStrategyPositionGroupBuyingPowerModelTests.cs

```diff
@@ -54,6 +58,259 @@ public void Setup()
+            new TestCaseData(OptionStrategyDefinitions.PutCalendarSpread, -20, -310 - -20, true),
+            new TestCaseData(OptionStrategyDefinitions.PutCalendarSpread, -20, -330 - -20, false).Explicit(),
+            new TestCaseData(OptionStrategyDefinitions.PutCalendarSpread, -20, 20, true),
+            new TestCaseData(OptionStrategyDefinitions.PutCalendarSpread, -20, 1000 - -20, true),
+            new TestCaseData(OptionStrategyDefinitions.PutCalendarSpread, -20, 10000 - -20, true),
+            new TestCaseData(OptionStrategyDefinitions.IronCondor, 0, 980, true),
+            new TestCaseData(OptionStrategyDefinitions.IronCondor, 0, 1000, false).Explicit(),
+            new TestCaseData(OptionStrategyDefinitions.IronCondor, 0, -980, true),
+            new TestCaseData(OptionStrategyDefinitions.IronCondor, 0, -1000, false).Explicit(),
+            new TestCaseData(OptionStrategyDefinitions.IronCondor, 20, 980 - 20, true),
+            new TestCaseData(OptionStrategyDefinitions.IronCondor, 20, 1000 - 20, false),
+            new TestCaseData(OptionStrategyDefinitions.IronCondor, 20, -20, true),
+            new TestCaseData(OptionStrategyDefinitions.IronCondor, 20, -980 - 20, true),
+            new TestCaseData(OptionStrategyDefinitions.IronCondor, 20, -1000 - 20, false).Explicit(),
+            new TestCaseData(OptionStrategyDefinitions.IronCondor, -20, -980 - -20, true),
+            new TestCaseData(OptionStrategyDefinitions.IronCondor, -20, -1000 - -20, false).Explicit(),
+            new TestCaseData(OptionStrategyDefinitions.IronCondor, -20, 20, true),
+            new TestCaseData(OptionStrategyDefinitions.IronCondor, -20, 980 - -20, true),
+            new TestCaseData(OptionStrategyDefinitions.IronCondor, -20, 1000 - -20, false).Explicit(),
+        };
+
+        [TestCaseSource(nameof(HasSufficientBuyingPowerForOrderTestCases))]
+        public void HasSufficientBuyingPowerForOrder(OptionStrategyDefinition optionStrategy, int initialPositionQuantity, int orderQuantity,
+            bool expectedResult)
+        {
+            _algorithm.SetCash(1000000);
+
+            IPositionGroup initialPositionGroup = null;
+            var templatePositionGroupQuantity = initialPositionQuantity;
+            if (initialPositionQuantity != 0)
+            {
+                initialPositionGroup = SetUpOptionStrategy(optionStrategy, initialPositionQuantity);
+            }
+            else
+            {
+                templatePositionGroupQuantity = 1;
+                initialPositionGroup = SetUpOptionStrategy(optionStrategy, templatePositionGroupQuantity);
+                foreach (var position in initialPositionGroup.Positions)
+                {
+                    var security = _algorithm.Securities[position.Symbol];
+                    security.Holdings.SetHoldings(0, 0);
+                }
+                Assert.AreEqual(0, _portfolio.PositionGroups.Count);
+            }
```

Comment:
Nice! Minor but maybe all of this can actually be handled by the `SetUpOptionStrategy` ?